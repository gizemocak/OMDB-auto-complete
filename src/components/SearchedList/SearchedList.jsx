import React from "react";
import { Link } from "react-router-dom";
import styled from "styled-components";

const SearchedListStyled = styled.div`
  line-height: 2.5;
  padding-left: 10%;
  padding-right: 10%;

  ul {
    list-style: none;
    text-decoration: none;
    text-align: left;
  }

  li {
    font-weight: bolder;
    border-bottom: 1px solid #5dbff0;
  }

  .movie-link {
    text-decoration: none;
    color: #3d576e;
    font-weight: bolder;
  }

  .movie-link:visited,
  .movie-link:active {
    color: #3d576e;
  }
`;

export default function SearchedList(props) {
  return (
    <SearchedListStyled>
      <ul>
        {props.movies.map(item => {
          return (
            <li key={item.imdbID}>
              <Link to={`/movie/${item.imdbID}`} className="movie-link">
                {item.Title}
              </Link>
            </li>
          );
        })}
      </ul>
    </SearchedListStyled>
  );
}
