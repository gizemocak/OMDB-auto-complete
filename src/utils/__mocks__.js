export const movieSearchResultsMock = [
  {
    Title: "Million Dollar Baby",
    Year: "2004",
    imdbID: "tt0405159",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BMTkxNzA1NDQxOV5BMl5BanBnXkFtZTcwNTkyMTIzMw@@._V1_SX300.jpg"
  },
  {
    Title: "Baby Driver",
    Year: "2017",
    imdbID: "tt3890160",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BMjM3MjQ1MzkxNl5BMl5BanBnXkFtZTgwODk1ODgyMjI@._V1_SX300.jpg"
  },
  {
    Title: "Gone Baby Gone",
    Year: "2007",
    imdbID: "tt0452623",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BYmM2NDNiNGItMTRhMi00ZDA2LTgzOWMtZTE2ZjFhMDQ2M2U5XkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
  },
  {
    Title: "Rosemary's Baby",
    Year: "1968",
    imdbID: "tt0063522",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BZmEwZGU2NzctYzlmNi00MGJkLWE3N2MtYjBlN2ZhMGJkZTZiXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
  },
  {
    Title: "The Boss Baby",
    Year: "2017",
    imdbID: "tt3874544",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BMTg5MzUxNzgxNV5BMl5BanBnXkFtZTgwMTM2NzQ3MjI@._V1_SX300.jpg"
  },
  {
    Title: "Bridget Jones's Baby",
    Year: "2016",
    imdbID: "tt1473832",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BMjQyMTI0MjIyOV5BMl5BanBnXkFtZTgwMzEyMjQ1ODE@._V1_SX300.jpg"
  },
  {
    Title: "Bringing Up Baby",
    Year: "1938",
    imdbID: "tt0029947",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BMmVkOTRiYmItZjE4NS00MWNjLWE0ZmMtYzg5YzFjMjMyY2RkXkEyXkFqcGdeQXVyNjc1NTYyMjg@._V1_SX300.jpg"
  },
  {
    Title: "Cry-Baby",
    Year: "1990",
    imdbID: "tt0099329",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BOGY3MzAyZDQtYTYyMS00ODI1LWEwYTEtMjU5MGFjZTYxZmQ4XkEyXkFqcGdeQXVyMjA0MzYwMDY@._V1_SX300.jpg"
  },
  {
    Title: "Baby",
    Year: "2015",
    imdbID: "tt3848892",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BYTdhNjBjZDctYTlkYy00ZGIxLWFjYTktODk5ZjNlMzI4NjI3XkEyXkFqcGdeQXVyMjY1MjkzMjE@._V1_SX300.jpg"
  },
  {
    Title: "What Ever Happened to Baby Jane?",
    Year: "1962",
    imdbID: "tt0056687",
    Type: "movie",
    Poster:
      "https://m.media-amazon.com/images/M/MV5BZmI0M2VmNTgtMWVhYS00Zjg1LTk1YTYtNmJmMjRkZmMwYTc2XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg"
  }
];
