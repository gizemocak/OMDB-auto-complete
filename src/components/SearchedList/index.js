import SearchedList from "./SearchedList";
export default SearchedList;
