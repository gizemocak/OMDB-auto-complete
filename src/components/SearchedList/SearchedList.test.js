import React from "react";
import renderer from "react-test-renderer";
import SearchedList from "./SearchedList";
import { MemoryRouter as Router } from "react-router-dom";
import { movieSearchResultsMock } from "../../utils/__mocks__.js";

test("should render SearchedList", () => {
  const props = {
    movies: movieSearchResultsMock
  };
  const component = renderer.create(
    <Router>
      <SearchedList {...props} />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
