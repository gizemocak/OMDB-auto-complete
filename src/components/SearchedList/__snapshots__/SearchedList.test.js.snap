// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SearchedList 1`] = `
<div
  className="sc-bdVaJa hNfqcs"
>
  <ul>
    <li>
      <a
        className="movie-link"
        href="/movie/tt0405159"
        onClick={[Function]}
      >
        Million Dollar Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt3890160"
        onClick={[Function]}
      >
        Baby Driver
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt0452623"
        onClick={[Function]}
      >
        Gone Baby Gone
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt0063522"
        onClick={[Function]}
      >
        Rosemary's Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt3874544"
        onClick={[Function]}
      >
        The Boss Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt1473832"
        onClick={[Function]}
      >
        Bridget Jones's Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt0029947"
        onClick={[Function]}
      >
        Bringing Up Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt0099329"
        onClick={[Function]}
      >
        Cry-Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt3848892"
        onClick={[Function]}
      >
        Baby
      </a>
    </li>
    <li>
      <a
        className="movie-link"
        href="/movie/tt0056687"
        onClick={[Function]}
      >
        What Ever Happened to Baby Jane?
      </a>
    </li>
  </ul>
</div>
`;
